src/main/java/org/cactoos/iterator/Sliced.java

```diff
@@ -52,22 +71,77 @@
      */
     public Sliced(final int start, final int count,
         final Iterator<T> iterator) {
-        this.iterator = new HeadOf<>(
-            count,
-            new Skipped<>(
-                start,
-                iterator
-            )
-        );
+        this(start, iterator, index -> index > start + count - 1);
+    }
+
+    /**
+     * Constructor.
+     * Constructs a head iterator with a number of heading elements
+     * @param iterator Decorated iterator
+     * @param count Maximum number of elements for resulted iterator
+     */
+    public Sliced(final Iterator<T> iterator, final int count) {
```

Comment:
@fanifieiev in cactoos, it is preferred to keep the iterator as the last argument of the constructor. I suppose you did this so that the two constructors does not conflict. In that case I think it is preferable to keep a simple API and remove this constructor: users will simply explicit the start parameter as 0.

Reply:
Done